src/parser/mod.rs

```diff
@@ -5256,14 +5256,19 @@ impl<'a> Parser<'a> {
 
     pub fn parse_create_trigger(
         &mut self,
+        or_alter: bool,
         or_replace: bool,
         is_constraint: bool,
     ) -> Result<Statement, ParserError> {
-        if !dialect_of!(self is PostgreSqlDialect | GenericDialect | MySqlDialect) {
+        if !dialect_of!(self is PostgreSqlDialect | GenericDialect | MySqlDialect | MsSqlDialect) {
             self.prev_token();
             return self.expected("an object type after CREATE", self.peek_token());
         }
 
+        if dialect_of!(self is MsSqlDialect) {
+            return self.parse_mssql_create_trigger(or_alter, or_replace, is_constraint);
+        }
```

Comment:
maybe this could be parsed via `self.dialect.parse_statement()` for the mssql version? since the dialect_of macro we try to stay away for new code

Reply:
That seems tricky. Can you help me understand what that code would look like? parse_create_trigger is happening inside a complicated branch of parse_create, so it seems like that logic from src/parser/mod.rs would need to be copy/pasted to src/dialect/mssql.rs which seems unfortunate.

Otherwise I don't know how to have the dialect's parse_statement implement custom logic only for "create trigger" but not for all the other "create xxx"'s

Reply:
ah yeah it would need to manually parse the alter, replace for the trigger, I agree that its not ideal but its probably the lesser evil, see [snowflake here for an example](https://github.com/apache/datafusion-sqlparser-rs/blob/main/src/dialect/snowflake.rs#L144)

Reply:
ah yeah it would need to manually parse the alter, replace for the trigger, I agree that its not ideal but its probably the lesser evil, see [snowflake here for an example](https://github.com/apache/datafusion-sqlparser-rs/blob/main/src/dialect/snowflake.rs#L144)

Reply:
ah yeah it would need to manually parse the alter, replace for the trigger, I agree that its not ideal but its probably the lesser evil, see [snowflake here for an example](https://github.com/apache/datafusion-sqlparser-rs/blob/main/src/dialect/snowflake.rs#L144)r